pkg/daemon/daemon.go

```diff
@@ -924,10 +800,52 @@ func (dn *Daemon) drainNode() error {
 		glog.Errorf("drainNode(): failed to drain node: %v", err)
 		return err
 	}
+	dn.enableUpdatingFlag()
 	glog.Info("drainNode(): drain complete")
 	return nil
 }
 
+func (dn *Daemon) getUpdatingFlag() bool {
+	dn.updatingFlagMutex.Lock()
+	defer dn.updatingFlagMutex.Unlock()
+	return dn.updatingFlag
+}
+
+func (dn *Daemon) enableUpdatingFlag() {
+	dn.updatingFlagMutex.Lock()
+	defer dn.updatingFlagMutex.Unlock()
+	dn.updatingFlag = true
+}
+
+func (dn *Daemon) disableUpdatingFlag() {
+	dn.updatingFlagMutex.Lock()
+	defer dn.updatingFlagMutex.Unlock()
+	dn.updatingFlag = false
+}
+
+func (dn *Daemon) installSignalHandler(signalCh chan struct{}) {
+	termChan := make(chan os.Signal, 2048)
+	signal.Notify(termChan, syscall.SIGTERM)
+
+	// Catch SIGTERM - if we're actively updating, we should avoid
+	// having the process be killed.
+	go func() {
+		for sig := range termChan {
+			//nolint:gocritic
+			switch sig {
+			case syscall.SIGTERM:
+				updateActive := dn.getUpdatingFlag()
+				if updateActive {
+					glog.Info("Got SIGTERM, but actively updating")
+				} else {
+					close(signalCh)
```

Comment:
what about using `stopCh`? is signalCh really needed?

Reply:
`stopCh` is an external source to stop the daemon, and `signalCh` is an internal way of stopping it. In other words, if we receive SIGTERM and we are not doing any drain/apply operation on the plugins, we need to stop the daemon.

Reply:
`stopCh` (as well as `exitCh`) is not written nor closed by anyone in production code. For sure we can even clean that in another PR, but at least use a different name for `installSignalHandler(...)` parameter.

Reply:
Renamed to `signalStopCh`